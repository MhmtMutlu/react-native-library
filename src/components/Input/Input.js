import React from 'react';
import {TextInput, View} from 'react-native';

import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import colors from '../../assets/styles/colors';

import styles from './Input.styles';

const Input = ({placeholder, value, onType, iconName, isSecure}) => {
  return (
    <View style={styles.container}>
      <TextInput
        style={styles.input}
        value={value}
        placeholder={placeholder}
        onChangeText={onType}
        secureTextEntry={isSecure}
        placeholderTextColor={colors.darkGrey2x}
      />
      <Icon name={iconName} size={25} color={colors.darkGrey2x} />
    </View>
  );
};

export default Input;
