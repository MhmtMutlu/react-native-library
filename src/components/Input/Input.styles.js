import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    paddingVertical: 3,
    paddingHorizontal: 10,
    margin: 10,
    backgroundColor: '#e0e0e0',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  input: {
    flex: 1,
  },
});
