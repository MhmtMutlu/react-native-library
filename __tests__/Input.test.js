import React from 'react';
import {render} from '@testing-library/react-native';
import Input from '../src/components/Input';

test('should match snapshot', () => {
  const comp = render(<Input />);
  expect(comp).toMatchSnapshot();
});

test('should render placeholder correctly', () => {
  const testPlaceholder = 'placeholder';
  const comp = render(<Input placeholder={testPlaceholder} />);
  const textInput = comp.getByTestId('text-input').props.placeholder;
  expect(textInput).toBe(testPlaceholder);
});
