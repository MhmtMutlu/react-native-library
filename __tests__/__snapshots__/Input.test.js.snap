// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#e0e0e0",
      "borderRadius": 10,
      "color": "#343a40",
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 5,
      "paddingHorizontal": 10,
      "paddingVertical": 1,
    }
  }
  testID="input-view"
>
  <TextInput
    multiline={false}
    numberOfLines={null}
    placeholderTextColor="#343a40"
    style={
      Object {
        "flex": 1,
      }
    }
    testID="text-input"
    textAlignVertical={null}
  />
</View>
`;

exports[`should render TextInputs css 1`] = `
Object {
  "flex": 1,
}
`;

exports[`should render Views css 1`] = `
Object {
  "alignItems": "center",
  "backgroundColor": "#e0e0e0",
  "borderRadius": 10,
  "color": "#343a40",
  "flexDirection": "row",
  "justifyContent": "center",
  "margin": 5,
  "paddingHorizontal": 10,
  "paddingVertical": 1,
}
`;
